src/test/java/io/github/wimdeblauwe/hsbt/mvc/PartialsController.java

```diff
@@ -0,0 +1,90 @@
+package io.github.wimdeblauwe.hsbt.mvc;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.List;
+
+@Controller
+public class PartialsController {
+
+    private final TodoRepository todoRepository;
+
+    public PartialsController(TodoRepository todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    public interface TodoRepository {
+        public int getNumberOfActiveItems();
+    }
+
+    public static class TodoItem{
+        String name;
+
+        public TodoItem(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+    }
+
+    @GetMapping("/partials/first")
+    public HtmxResponse getFirstPartials() {
+        return new HtmxResponse().addTemplate("users :: list");
+    }
+
+    @GetMapping("/partials/main-and-partial")
+    public HtmxResponse getMainAndPartial(Model model) {
+        model.addAttribute("userCountOob", true);
+        model.addAttribute("userCount", 5);
+        return new HtmxResponse()
+                .addTemplate("users :: list")
+                .addTemplate("users :: count");
+    }
+
+    @PostMapping("/partials/add-todo")
+    public HtmxResponse htmxAddTodoItem(TodoItem item, Model model) {
+        model.addAttribute("item", item);
+        model.addAttribute("itemCountSwap", "true");
+        model.addAttribute("numberOfActiveItems", todoRepository.getNumberOfActiveItems());
+        return new HtmxResponse()
+                .addTemplate("fragments :: todoItem")
+                .addTemplate("fragments :: active-items-count");
+    }
+
+    @GetMapping("/partials/triggers")
+    public HtmxResponse getPartialsAndTriggers(Model model) {
+        model.addAttribute("userCountOob", true);
+        model.addAttribute("userCount", 5);
+
+        return new HtmxResponse().addTemplate("users :: list")
+                .addTemplate("users :: count")
+                .addTrigger("usersCounted")
+                .addTrigger("usersCountedSwap", "swap detail", HxTriggerLifecycle.SWAP)
+                .addTrigger("usersCountedSettle1", "aDetail", HxTriggerLifecycle.SETTLE)
+                .addTrigger("usersCountedSettle2", null, HxTriggerLifecycle.SETTLE);
+    }
+
+
+    @GetMapping("/partials/extension")
+    public HtmxResponse getPartialsViaExtension(Model model) {
+        MyHtmxResponse response = new MyHtmxResponse(model);
```

Comment:
This is an example of subclassing the `HtmxResponse` and being able to extend it (note the `sendAlertPartial` that sets model content and templates)

The subclassing only works as long as the response type is `HtmxResponse` though. So if the signature was `public MyHtmxRespose getPartials....` it doesn't work.

Reply:
As stated above, I am no fan of the subclassing. Using my proposal above, this could be written as:

```
HtmxResponse response = new HtmxResponse();
response.addResponse(MyApplicationHtmxResponseHelper.sendAlertPartial(model, "Warning!"));
model.addAttribute("userCountOob", true);
model.addAttribute("userCount", 5);
response.addTemplate("users :: list")
                .addTemplate("users :: count");
return response;
```

This would require a new `addResponse(HtmxResponse response)` method on `HtmxResponse` that would combine 2 response objects. (Maybe use `combineWithResponse` as method name instead of `addResponse`?)